.github/workflows/tests.yml

```diff
@@ -26,6 +26,7 @@ jobs:
   bully-build:
     name: Bully build
     runs-on: self-hosted
+    needs: test
```

Comment:
This is unrelated to the `opt-level` change, but seems to make CI run better (prevents some kind of race causing a file to be deleted which is still needed by other tests).

Reply:
Hmm, that's interesting.
I suspect that, unfortunately, this will not solve the problem. Two or more concurrent "test" jobs can possibly produce the same issue.
Let me explain: each job is launched on random free runner (1 of 16), so each of them have it's own isolated workdir. Except only moment - they still share `target/release` and `target/debug` directories (you can see that I made them as symlinks), and this directories are shared for all runs and all jobs (even concurrent). I did a little research before doing it, it was stated that cargo `target` dir is OK for concurrent builds, but after seeing this - I doubt a lot. I mean, it's impossible to explain those races another way (because those are only files that are shared/concurrently used). I should probably redesign this part...
Do you know which particular file was under race? If it's under `target/release` or `target/debug` - it's definitely my fault here :)